homeassistant/components/bmw_connected_drive/quality_scale.yaml

```
@@ -0,0 +1,71 @@
+rules:
+  # Bronze
+  action-setup: done
+  appropriate-polling: done
+  brands: done
+  common-modules: done
+  config-flow-test-coverage: done
+  config-flow: done
+  dependency-transparency: done
+  docs-actions: done
+  docs-high-level-description: done
+  docs-installation-instructions: done
+  docs-removal-instructions: todo
+  entity-event-setup:
+    status: exempt
+    comment: |
+      This integration doesn't have any events.
+  entity-unique-id: done
+  has-entity-name: done
+  runtime-data: done
```

```suggestion
  runtime-data: todo
```
there are still a lot of untyped `ConfigEntry` (e.g. init, coordinator)

---

Not sure what you mean by "a lot".
`entry.runtime_data` is typed (https://github.com/home-assistant/core/blob/dev/homeassistant/components/bmw_connected_drive/__init__.py#L144 and https://github.com/home-assistant/core/blob/dev/homeassistant/components/bmw_connected_drive/__init__.py#L52-L59)

I do see that the coordinator itself is not fully typed (#132087)